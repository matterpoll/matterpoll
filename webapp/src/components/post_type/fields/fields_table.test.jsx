import React from 'react';
import {shallow} from 'enzyme';

import FieldsTable from 'components/post_type/fields/fields_table';

describe('components/post_type/fields/FiledsTable', () => {
    const baseProps = {
        attachment: {
            fields: [
                {
                    title: 'title1',
                    value: 'value1',
                    short: false,
                },
            ],
        },
        options: {
            mentionHighlight: false,
            markdown: false,
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<FieldsTable {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot without any fields', () => {
        const newProps = {
            ...baseProps,
        };
        newProps.attachment.fields = [];
        const wrapper = shallow(<FieldsTable {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with two fields', () => {
        const newProps = {
            ...baseProps,
            attachment: {
                fields: [
                    {title: 'title1', value: 'value1', short: false},
                    {title: 'title2', value: 'value2', short: false},
                ],
            },
        };
        const wrapper = shallow(<FieldsTable {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with a short field', () => {
        const newProps = {
            ...baseProps,
            attachment: {
                fields: [
                    {title: 'title1', value: 'value1', short: true},
                ],
            },
        };
        const wrapper = shallow(<FieldsTable {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with two short fields', () => {
        const newProps = {
            ...baseProps,
            attachment: {
                fields: [
                    {title: 'title1', value: 'value1', short: true},
                    {title: 'title2', value: 'value2', short: true},
                ],
            },
        };
        const wrapper = shallow(<FieldsTable {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with three short fields', () => {
        const newProps = {
            ...baseProps,
            attachment: {
                fields: [
                    {title: 'title1', value: 'value1', short: true},
                    {title: 'title2', value: 'value2', short: true},
                    {title: 'title3', value: 'value3', short: true},
                ],
            },
        };
        const wrapper = shallow(<FieldsTable {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
