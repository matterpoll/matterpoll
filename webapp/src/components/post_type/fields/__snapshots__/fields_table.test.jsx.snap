// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_type/fields/FiledsTable should match snapshot 1`] = `
<div>
  <table
    className="attachment-fields"
    key="attachment__table__0"
  >
    <thead>
      <tr />
    </thead>
    <tbody>
      <tr />
    </tbody>
  </table>
  <table
    className="attachment-fields"
    key="attachment__table__1"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-0__1"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title1))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-0__1"
        >
          mockMessageHtmlToComponent(mockFormatText(value1))
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/post_type/fields/FiledsTable should match snapshot with a short field 1`] = `
<div>
  <table
    className="attachment-fields"
    key="attachment__table__0"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-0__0"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title1))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-0__0"
        >
          mockMessageHtmlToComponent(mockFormatText(value1))
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/post_type/fields/FiledsTable should match snapshot with three short fields 1`] = `
<div>
  <table
    className="attachment-fields"
    key="attachment__table__0"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-0__0"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title1))
        </th>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-1__0"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title2))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-0__0"
        >
          mockMessageHtmlToComponent(mockFormatText(value1))
        </td>
        <td
          className="attachment-field"
          key="attachment__field-1__0"
        >
          mockMessageHtmlToComponent(mockFormatText(value2))
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="attachment-fields"
    key="attachment__table__1"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-2__1"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title3))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-2__1"
        >
          mockMessageHtmlToComponent(mockFormatText(value3))
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/post_type/fields/FiledsTable should match snapshot with two fields 1`] = `
<div>
  <table
    className="attachment-fields"
    key="attachment__table__0"
  >
    <thead>
      <tr />
    </thead>
    <tbody>
      <tr />
    </tbody>
  </table>
  <table
    className="attachment-fields"
    key="attachment__table__1"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-0__1"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title1))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-0__1"
        >
          mockMessageHtmlToComponent(mockFormatText(value1))
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="attachment-fields"
    key="attachment__table__2"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-1__2"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title2))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-1__2"
        >
          mockMessageHtmlToComponent(mockFormatText(value2))
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/post_type/fields/FiledsTable should match snapshot with two short fields 1`] = `
<div>
  <table
    className="attachment-fields"
    key="attachment__table__0"
  >
    <thead>
      <tr>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-0__0"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title1))
        </th>
        <th
          className="attachment-field__caption"
          key="attachment__field-caption-1__0"
          width="50%"
        >
          mockMessageHtmlToComponent(mockFormatText(title2))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="attachment-field"
          key="attachment__field-0__0"
        >
          mockMessageHtmlToComponent(mockFormatText(value1))
        </td>
        <td
          className="attachment-field"
          key="attachment__field-1__0"
        >
          mockMessageHtmlToComponent(mockFormatText(value2))
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/post_type/fields/FiledsTable should match snapshot without any fields 1`] = `""`;
